src/main/java/com/github/rodionovsasha/commentservice/services/impl/TopicServiceImpl.java

```diff
@@ -0,0 +1,59 @@
+package com.github.rodionovsasha.commentservice.services.impl;
+
+import com.github.rodionovsasha.commentservice.entities.Topic;
+import com.github.rodionovsasha.commentservice.exceptions.TopicAccessException;
+import com.github.rodionovsasha.commentservice.exceptions.TopicNotFoundException;
+import com.github.rodionovsasha.commentservice.repositories.TopicRepository;
+import com.github.rodionovsasha.commentservice.services.TopicService;
+import com.github.rodionovsasha.commentservice.services.UserService;
+import lombok.AllArgsConstructor;
+import lombok.val;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.data.domain.Sort;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+@AllArgsConstructor
+@Service @Transactional
+public class TopicServiceImpl implements TopicService {
+    private final TopicRepository repository;
+    private final UserService userService;
+
+    @Override
+    public Topic start(String title, long userId) {
+        return repository.save(new Topic(title, userService.getActiveUser(userId)));
+    }
+
+    @Override
+    public void archive(long topicId, long userId) {
+        userService.checkUserActive(userId);
+        val topic = getById(topicId);
+        if (topic.getOwner().getId() != userId) {
+            throw TopicAccessException.forId(userId);
+        }
+        repository.delete(topicId);
```

Comment:
Archive should not delete topic, it should just set a flag `archived`(we need to add this flag to the model) to true

Reply:
yes, forget..